public/app/features/correlations/CorrelationsPage.tsx

```diff
@@ -54,61 +56,59 @@ export default function CorrelationsPage() {
     get: { execute: fetchCorrelations, ...get },
   } = useCorrelations();
 
-  useEffect(() => {
-    fetchCorrelations();
-    // we only want to fetch data on first render
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, []);
-
   const canWriteCorrelations = contextSrv.hasPermission(AccessControlAction.DataSourcesWrite);
 
   const handleAdded = useCallback(() => {
     reportInteraction('grafana_correlations_added');
-    fetchCorrelations();
+    fetchCorrelations({ page: page.current });
     setIsAdding(false);
   }, [fetchCorrelations]);
 
   const handleUpdated = useCallback(() => {
     reportInteraction('grafana_correlations_edited');
-    fetchCorrelations();
+    fetchCorrelations({ page: page.current });
   }, [fetchCorrelations]);
 
   const handleDelete = useCallback(
-    (params: RemoveCorrelationParams) => {
-      remove.execute(params);
+    async (params: RemoveCorrelationParams, isLastRow: boolean) => {
+      await remove.execute(params);
+      reportInteraction('grafana_correlations_deleted');
+
+      if (isLastRow) {
+        page.current--;
+      }
+      fetchCorrelations({ page: page.current });
     },
-    [remove]
+    [remove, fetchCorrelations]
   );
 
-  // onDelete - triggers when deleting a correlation
-  useEffect(() => {
-    if (remove.value) {
-      reportInteraction('grafana_correlations_deleted');
-    }
-  }, [remove.value]);
-
   useEffect(() => {
-    if (!remove.error && !remove.loading && remove.value) {
-      fetchCorrelations();
-    }
-  }, [remove.error, remove.loading, remove.value, fetchCorrelations]);
+    fetchCorrelations({ page: page.current });
+  }, [fetchCorrelations]);
 
   const RowActions = useCallback(
     ({
       row: {
+        index,
         original: {
           source: { uid: sourceUID, readOnly },
           uid,
         },
       },
-    }: CellProps<CorrelationData, void>) =>
-      !readOnly && (
-        <DeleteButton
-          aria-label="delete correlation"
-          onConfirm={() => handleDelete({ sourceUID, uid })}
-          closeOnConfirm
-        />
-      ),
+    }: CellProps<CorrelationData, void>) => {
+      return (
+        !readOnly && (
+          <DeleteButton
+            aria-label="delete correlation"
+            onConfirm={() =>
+              handleDelete({ sourceUID, uid }, page.current > 0 && index === 0 && data?.correlations.length === 1)
```

Comment:
Shouldn't this be 1 not 0? The page is always  > 0. At the moment if a very last correlation is removed, and a new one is created, then no page is selected in Pagination component.

```suggestion
              handleDelete({ sourceUID, uid }, page.current > 1 && index === 0 && data?.correlations.length === 1)
```

Reply:
Good catch! Tested by deleting all correlations manually then created one, it works as expected.

Reply:
and that's why folks reviews are important <3 thanks for catching this!